src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.ComposeCache.targets

```diff
@@ -59,15 +59,20 @@ Copyright (c) .NET Foundation. All rights reserved.
       <ComposeDir Condition="'$(ComposeDir)' == ''">$(DefaultComposeDir)</ComposeDir>
       <ComposeDir Condition="'$(DoNotDecorateComposeDir)' != 'true'">$([System.IO.Path]::Combine($(ComposeDir), $(_TFM)))</ComposeDir>
       <PublishDir>$([System.IO.Path]::GetFullPath($(ComposeDir)))</PublishDir>
-      <_CrossProjFileDir>$([System.IO.Path]::Combine($(MSBuildProjectExtensionsPath),"Optimize"))</_CrossProjFileDir>
+      <TEMP Condition="'$(TEMP)' == ''">$([System.IO.Path]::GetTempPath())</TEMP>
```

Comment:
dotnet-cache does is not like other dotnet commands, as it is not associated with a folder. Think of it like install command.

However an option is provided for the user to specify the intermediate directory.